src/main/java/org/springframework/data/tarantool/core/mapping/BasicTarantoolPersistentEntity.java

```
@@ -32,4 +44,82 @@ public String getSpaceName() {
+        }
+        return getType().getAnnotation(TarantoolIdClass.class).value();
+    }
+
+
+    @Override
+    protected TarantoolPersistentProperty returnPropertyIfBetterIdPropertyCandidateOrNull(TarantoolPersistentProperty property) {
+        if (hasTarantoolIdClassAnnotation()) {
+            // id accessor will be used to access multiple id's
+//            return null;
+            return property.isIdProperty() ? property : null;
+        }
+        return super.returnPropertyIfBetterIdPropertyCandidateOrNull(property);
+    }
+
+    @Override
+    public IdentifierAccessor getIdentifierAccessor(Object bean) {
+        if (hasTarantoolIdClassAnnotation()) {
+            return new TarantoolCompositeIdentifierAccessor(this, bean);
+        }
+        return super.getIdentifierAccessor(bean);
+    }
+
+    private static class TarantoolCompositeIdentifierAccessor extends TargetAwareIdentifierAccessor {
+
+        private final BasicTarantoolPersistentEntity<?> entity;
+        private final Object bean;
+
+        TarantoolCompositeIdentifierAccessor(BasicTarantoolPersistentEntity<?> entity, Object bean) {
+            super(bean);
+            this.entity = entity;
+            this.bean = bean;
+        }
+
+        //FIXME
+        @SneakyThrows
+        @Override
+        public Object getIdentifier() {
+            Class<?> idClass = entity.getTarantoolIdClass();
+            Assert.notNull(idClass, "ID class is null for entity " + entity);
+
+            if (idClass.isInstance(bean)) {
+                return bean;
+            }
+            Object id = Objects.requireNonNull(idClass).newInstance();
+
+
+            for (Field field : idClass.getDeclaredFields()) {
+                if (!Modifier.isStatic(field.getModifiers())) {
+                    TarantoolPersistentProperty property = entity.getPersistentProperty(field.getName());
+
+                    Assert.notNull(property, "Property '"+ field.getName() + "' is null for " + entity);
```

I'm not sure that we should throw an error here. It will possibly prohibit inheritance for the entity classes.

---

I think we should keep it - it's not clean how the composite id will work. 
Need to investigate this case - I'll add test with property inheritance later.